packages/data-broker/src/DataBroker.js

```diff
@@ -36,15 +36,13 @@ export class DataBroker {
   }
 
   async fetchDataSources(dataSourceSpec) {
-    const errorMessage = 'Please provide at least one existing data source code';
-
     const { code } = dataSourceSpec;
     if (!code || (Array.isArray(code) && code.length === 0)) {
-      throw new Error(errorMessage);
+      throw new Error('Please provide at least one existing data source code');
     }
-    const dataSources = await this.models.dataSource.findOrDefault(dataSourceSpec);
+    const dataSources = await this.models.dataSource.find(dataSourceSpec);
     if (dataSources.length === 0) {
-      throw new Error(errorMessage);
+      throw new Error(`None of the following data sources exist: ${code}`);
```

Comment:
Great suggestion, we should definitely log a warning! It may be that we it's valid to throw an error since one will be thrown anyway - I'll confirm this

Reply:
So I did have a look, DHIS throws an error if **none** of the requested `dx` elements exist, if at least one exists the rest are ignored. I also believe that internal data fetching will just ignore all missing elements, but I didn't confirm this.

Based on the above, I suggest logging useful warnings for missing data elements, but not throw an error